expat/CMakeLists.txt

```diff
@@ -324,6 +325,40 @@ if(BUILD_tests)
     add_test(runtestspp tests/runtestspp)
 endif(BUILD_tests)
 
+if(BUILD_fuzzers)
+    if(NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        message(SEND_ERROR
+            "Building fuzz targets without clang is not supported. Please set "
+            "-DCMAKE_C_COMPILER=clang.")
+    endif()
+
+    string(FIND "${CMAKE_EXE_LINKER_FLAGS}" "-fsanitize" sanitizer_present)
+    if(${sanitizer_present} EQUAL "-1")
+        message(WARNING
+            "There was no sanitizer present when building the fuzz targets. "
+            "This is likely in error - consider adding "
+            "-DCMAKE_C_FLAGS='-fsanitize=<sanitizer>' and "
+            "-DCMAKE_EXE_LINKER_FLAGS='-fsanitize=<sanitizer>' to your cmake "
+            "execution.")
+    endif()
+
+    set(encoding_types UTF-16 UTF-8 ISO-8859-1 US-ASCII UTF-16BE UTF-16LE)
+    set(fuzz_targets xml_parse_fuzzer xml_parsebuffer_fuzzer)
+
+    foreach(fuzz_target ${fuzz_targets})
+        foreach(encoding_type ${encoding_types})
+            set(target_name ${fuzz_target}_${encoding_type})
+            add_executable(${target_name} fuzz/${fuzz_target}.c ${expat_SRCS})
```

Comment:
@hctim I notice now: If we change that line to `target_link_libraries(${target_name} expat)` we'd no longer compile 3 C files 7 times but once. Are you aware of any effect of `-fsanitize=fuzzer-no-link` that we need to recompile these files for, as well? Many thanks!

Reply:
Yes, you'll need to compile with `-fsanitize=fuzzer-no-link` in order to generate fuzzer coverate instrumentation across the entire library.

Reply:
I see! So we need to compile Expat more than once… but maybe *twice* is enough if all fuzzers link to a single static library? If you have a minute for review, I have a demo of that idea at #335. Thanks!